schemas/client.yml

```diff
@@ -59,34 +59,91 @@
+        decides which value to use.
+        
+        In the case of fully and partially qualified domain names it is recommended to parse out 
+        the additional fields of top_level_domain, registered_domain, subdomain, and hostname
+        as they are available.
+      example: www.east.mydomain.co.uk 
+
+    # Domain Stuff
+    - name: hostname
+      level: core
+      type: keyword
+      short: Hostname of the system.
+      description: >
+        Hostname of the system.
+
+        It normally contains what the `hostname` command returns on the host machine, or the
+        host portion of a fully qualified domain name.
+        
+        For example, the hostname portion of "www.east.mydomain.co.uk " is "www".
+      example: www
+    
     - name: registered_domain
       level: extended
       type: keyword
-      short: The highest registered client domain, stripped of the subdomain.
+      short: The highest registered domain, stripped of the subdomain.
       description: >
-        The highest registered client domain, stripped of the subdomain.
+        The highest registered domain, stripped of the subdomain.
 
-        For example, the registered domain for "foo.google.com" is "google.com".
+        For example, the registered domain for " www.east.mydomain.co.uk " is "mydomain.co.uk".
 
         This value can be determined precisely with a list like the public
         suffix list (http://publicsuffix.org). Trying to approximate this by
         simply taking the last two labels will not work well for TLDs such as "co.uk".
-      example: google.com
+      example: mydomain.co.uk
 
     - name: top_level_domain
       level: extended
       type: keyword
       short: The effective top level domain (com, org, net, co.uk).
       description: >
         The effective top level domain (eTLD), also known as the domain suffix,
-        is the last part of the domain name.
-        For example, the top level domain for google.com is "com".
+        is the last part of the domain name and is typically inclusive of the 
+        top level domain (e.g. "co"), as well as including country code and region codes
+        For example, the top level domain for www.east.mydomain.co.uk is "co.uk".
 
         This value can be determined precisely with a list like the public
-        suffix list (http://publicsuffix.org). Trying to approximate this by
-        simply taking the last label will not work well for effective TLDs such as "co.uk".
+        suffix list (http://publicsuffix.org). 
       example: co.uk
 
+    - name: subdomain
+      level: extended
+      type: keyword
+      short: The subdomain of the domain.
+      description: >
+        The subdomain portion of a fully qualified domain name includes all of the names except 
+        the host name under the registered_domain.  In a partially qualified domain, or if the 
+        the qualification level of the full name cannot be determined, subdomain contains all of
```

Comment:
I'm not sure of excluding the "hostname" from the subdomain, or assuming it should populate the hostname field.

What happens when the the "hostname" is not the hostname of the machine as with most publicly facing URLs?

For example mail.google.com is unlikely to be a machine whose actual hostname is "mail". It could be more than one machine load balanced where an individual host responded, or more than one subdomain hosted by the same machine. Is this what ``..or if the qualification level of the full name cannot be determined`` is supposed to cover? There seem to be different schools of thought on whether the leftmost part of a domain counts as a hostname, or always a subdomain, or sometimes one or the other depending on the context that can't be determined from just looking at the url.

This is particularly important in fields set groups outside url and dns (like client and server) where you might be recording the machines actual hostname based on network traffic and not the url, but copying from url.domain to server.domain. If you drop mail from the subdomain but have hostname as the real internal IP based one the information is inconsistent.

Reply:
So this one is annoyingly specific to a number of criteria... 


- Locality (local hostname vs. host portion of a fqdn)
- Record Type (CName(s) vs A record)
- Truth (dns "host"/"A Record" that resolves to a load balancer, vs round robin dns load balancing to multiple individual hosts)

I was trying to adjust, with minimal breaks, dns info across multiple fields and I think the valid concerns come down to frame of reference.

Organizationally remote from DNS Server:
in the mail.google.com example - I get a CName to googlemail.l.google.com. which resolves to a single IP (as far as I can tell from 1 nslookup) - so regardless of the backend reality, final dns response looks like host.registered_domain.  (theres also no issue with multiple responses mapping the same name to multiple ips (or vice versa)


I'm thinking maybe pulling name/hostname out of the normalization